### PR TITLE
Add call metadata to events

### PR DESCRIPTION
**Pull request by amitaekbote**
Add call metadata information to events

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java [hunk 1]
```diff
@@ -608,15 +608,16 @@ public Completable recordTaskPlacement(String taskId, Function<Task, Task> chang
         Task changed = modelUpdateEvent.getChangedEntityHolder().getEntity();
         if (!modelUpdateEvent.getPreviousEntityHolder().isPresent()) {
             return tasksPredicate.test(Pair.of(job, changed))
-                    ? Optional.of(toNewTaskUpdateEvent(job, changed))
+                    ? Optional.of(toNewTaskUpdateEvent(job, changed, modelUpdateEvent.getCallMetadata()))
                     : Optional.empty();
         }
         Task previous = modelUpdateEvent.getPreviousEntityHolder().get().getEntity();
+        CallMetadata callMetadata = (CallMetadata)modelUpdateEvent.getChangedEntityHolder().getAttributes().get(JobManagerConstants.JOB_MANAGER_ATTRIBUTE_CALLMETADATA);
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java]: Why the `CallMetadata` value is not taken directly from the modelUpdateEvent like above?

**Comment by tbak** [comment]
After the failover all job and task records will miss the `CallMetadata` attributes. Is there an assumption that this attribute is always present, or it is optional and if missing `undefined` value is used?

File: titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java (hunk 1 above)
Comment (amitaekbote) [reply titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java]: I must have changed that before I used modelUpdateEvent, thanks for catching that.

**Comment by amitaekbote** [comment]
I currently assume its always present, I will check in the event getCallMetadata for sanity

**Pull request merged**